Fix linux build and QT Deprecations

contra-bit opened the pull request:
This fixes #3292 and subsequent errors mentioned in the issue.
Also this fixes most QT deprecation warnings.
 
Please merge,

contra-bit committed:
set gcc12 as the default compiler for linux

contra-bit committed:
update documentation to include info about gcc12 for building with linux

contra-bit committed:
update .gitignore to ignore build artefacts

contra-bit committed:
Include QTQuick in the build

contra-bit committed:
fix unsued return value error while loading qt translations

contra-bit committed:
fix rb_data_object_allow deprecation

contra-bit committed:
fix deprecation warning for QApplication::setActiveWindow(QWidget*)

contra-bit committed:
Do not set default attributes

samaaron commented:
@contra-bit - thanks so much for this. One question I have is why including QtQuick was necessary? I'm pretty sure it's not used. Perhaps it pulls in deps which are? (In which case it might be better to be more specific rather than relying on transitive deps?)

@lilyinstarlight - do you have any comments on these commits?

contra-bit committed:
Revert "Include QTQuick in the build"

This reverts commit 4280ab4033bdc773b6f4a26038835fb993499b35.

contra-bit commented:
Good call, I just removed the QtQuick and could still build.

Prior I added QtQuck, due to this error: QQuickWindow: No such file or directory.


Please note, that I have reverted the addition of QtQuick.
Thank you for the quick review.

samaaron committed:
Spider - update active support to 7.0.6

Switches transient gems - thread_safe and atomic are replaced with concurrent_ruby. tzinfo is also pulled in.

samaaron commented:
@contra-bit thanks for checking and reverting the QtQuick addition.

One other thing is that I'm a bit wary of patching upstream deps (as you have done with atomic). I've therefore taken this opportunity to update active_support to the latest version which drops thread_safe/atomic in preference for concurrent_ruby. I'm therefore interested to know if this fixes/messes things up for you in linux land. I'm hoping it should be fine...

https://github.com/sonic-pi-net/sonic-pi/commit/f76075139640165c6309c440879fbaa21603f8f8

samaaron committed:
Spider - improve runtime error reporting

Recent Ruby versions introduced a detailed_message method to Exception which includes the useful "did you mean" information which was dropped from the message method.

contra-bit commented:
I can confirm, that removing atomic_reference.c  does not break the build :)

samaaron commented:
@contra-bit - great. Any chance you could fix the conflict so I can run the CI tests?

contra-bit committed:
Merge remote-tracking branch 'upstream/dev' into dev

contra-bit commented:
Yes. Thanks for asking. I overlooked that merge conflict. 


I am so gratefull for your work. Thaks alot.  I learn so much about sonic pi and tau this week. I appreciate your time and effort, espcially your fast and high quality responses :heart: :headphones:  :tada: :pie:

contra-bit committed:
delete atomic_reference.c

contra-bit commented:
This resolve was weird (0 bits changed) .  I manually deleted atomic_reference.c, please check if there are any other files, which have reappeared.

samaaron commented:
@contra-bit - thanks so much for this - hugely appreciated!
